FIX ElasticaService::define() now allows index to be deleted before (re)defining it again

robbieaverill opened the pull request:
This fixes a bug where changing mapping types on a DataObject after the index has been created would result in a fatal error, but allowing you to explicitly delete the index before redefining it again.

Fixes https://github.com/heyday/silverstripe-elastica/issues/19

robbieaverill committed:
FIX ElasticaService::define() now allows index to be deleted before (re)defining it again

This fixes a bug where changing mapping types on a DataObject after the index
has been created would result in a fatal error, but allowing you to explicitly
delete the index before redefining it again.

sminnee commented on composer.json:
```diff
@@ -36,9 +36,13 @@
         "ruflin/elastica": "~5.3",
         "symbiote/silverstripe-queuedjobs": "~4.0"
     },
+    "require-dev": {
```
Should we add a .travis.yml while we're at it? ;-)

robbieaverill replied:
Done, example build via our fork here: https://travis-ci.org/creative-commoners/silverstripe-elastica/builds/515513180

sminnee reviewed (commented):
1 optional suggestion

robbieaverill committed:
Add Travis build configuration, phpcs ruleset and phpunit test suite configuration

robbieaverill committed:
FIX Remove PHP 7 specific code syntax to restore PHP 5.6 compatibility

PHP 5.6 is listed as the minimum requirement in composer.json, this restores that.
Fixes https://github.com/heyday/silverstripe-elastica/issues/15

robbieaverill committed:
API Remove PHP 7 return type hint

This is a breaking change, but the method is proteted. Up to the maintainers as
to whether they are happy to merge this. We can also bump the minimum version
of PHP to ^7.0 instead.

stevie-mayhew commented:
@tractorcow do you mind giving this a review when you have time?

tractorcow commented:
Looks good to me.

robbieaverill commented:
Thanks!
